### DISCUSSION TITLE
Moving from OCPP v1.6 to v2.0

### DISCUSSION THREAD

**Discussion by ezolti** [ChargeTimeEU/Java-OCA-OCPP#322] [Q&A] [answered]
Hi
I am building a charging center.
When using v1.6 I used a class that implements ServerCoreEventHandler which handles all Requests (8 of them).
like handleBootNotificationRequest , handleMeterValuesRequest.
Now in v2.0, I can see Feature that need to be used but only 2 handlers
IServerBootNotificationRequestHandler and IServerStatusNotificationRequestHandler
What am I missing?
What do I need to do?
Thanks

**Comment by TVolden** [comment]
Hi @ezolti,

Thank you for using this new discussion section to ask you question.

As far as I can tell you should implement the ServerProvisioningEventHandler:

https://github.com/ChargeTimeEU/Java-OCA-OCPP/blob/master/ocpp-v2/src/main/java/eu/chargetime/ocpp/v201/feature/function/ServerProvisioningEventHandler.java

Please note there is two version 2 modules at this moment, the one I started way back with a new architecture (where handlers were part of the feature, to simplify iterative implementations) and the new `v2` which supports multiple protocol versions. You should use `v2`.

**Reply by ezolti** [reply]
I am using
	<dependency>
		<groupId>com.github.ChargeTimeEU.Java-OCA-OCPP</groupId>
		<artifactId>common</artifactId>
		<version>v1.2</version>
	</dependency>
	<dependency>
		<groupId>com.github.ChargeTimeEU.Java-OCA-OCPP</groupId>
		<artifactId>OCPP-J</artifactId>
		<version>v1.2</version>
	</dependency>
	<dependency>
		<groupId>com.github.ChargeTimeEU.Java-OCA-OCPP</groupId>
		<artifactId>v2_0</artifactId>
		<version>v1.2</version>
	</dependency>
The error is : The import eu.chargetime.ocpp.v201 cannot be resolved
which dependency to use ?

**Reply by ezolti** [reply]
I use mvn install and use it on my local repository
	<dependency>
    	<groupId>eu.chargetime.ocpp</groupId>
    	<artifactId>v1_6</artifactId>
    	<version>1.2.0</version>
	</dependency>
	<dependency>
    	<groupId>eu.chargetime.ocpp</groupId>
    	<artifactId>common</artifactId>
    	<version>1.2.0</version>
	</dependency>
	<dependency>
    	<groupId>eu.chargetime.ocpp</groupId>
    	<artifactId>ocpp-v2</artifactId>
    	<version>1.2.0</version>
	</dependency>

**Comment by ezolti** [comment]
@TVolden 
In ServerProvisioningEventHandler there are 3
handleBootNotificationRequest
handleHeartbeatRequest
handleNotifyReportRequest
What should I do with the rest which are in use at OCPP v1.6?
handleMeterValuesRequest
handleStartTransactionRequest
handleStopTransactionRequest
and others.
Thanks

**Reply by ezolti** [reply]
@TVolden 
Can you please point to the right way to go?

**Reply by TVolden** [reply]
Hi @ezolti,

I'm not sure, I haven't implemented this part. The specification kinda dropped the idea of feature profiles, so it a little more messy now.

Take a look in: https://github.com/ChargeTimeEU/Java-OCA-OCPP/tree/master/ocpp-v2/src/main/java/eu/chargetime/ocpp/v201/feature/function

There is a ServerTransactionEventHandler interface, I think that may be it.

**Reply by ezolti** [reply, marked as answer]
Thanks @TVolden
